### DISCUSSION TITLE
Actions broken-link.yml fails: too many redirects

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [alshedivat/al-folio#2366] [Q&A]
### Have you checked that your issue isn't already filed?

- [X] I read through [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md) and searched through the [past issues](https://github.com/alshedivat/al-folio/issues), none of which addressed my issue.
- [X] Yes, I have checked that this issue isn't already filed.

### Bug description

Actions broken-link.yml fails

### How to reproduce the bug

Put a link to https://www.kth.se/profile/andbr in some post. For example, I put it [here](https://github.com/raffaem/raffaem.github.io/blob/6e863bdcb6b7f78b1a8d10f379bba51dd7c06170/_posts/2023-07-03-DRUID23.md?plain=1#L11).

### Error messages and logs

```
[ERR] https://www.kth.se/profile/andbr | Failed: Too many redirects: error following redirect for url (https://www.kth.se/files/auth/silent/bounce?nextUrl=https%3A%2F%2Fwww.kth.se%2Fprofile%2Fandbr): too many redirects
```

See [here](https://github.com/raffaem/raffaem.github.io/actions/runs/8795624976) for full log.

### What operating system are you using?

Linux

### Where are you seeing the problem on?

Deployed site

**Comment by george-gca** [comment]
As you can see in the error message, it states:

```
Failed: Too many redirects: error following redirect for url (https://www.kth.se/files/auth/silent/bounce?nextUrl=https%3A%2F%2Fwww.kth.se%2Fprofile%2Fandbr): too many redirects
```

Your page for some reason redirects too many times. The broken links workflow uses [lychee-action](https://github.com/lycheeverse/lychee-action), which in turn uses [lychee](https://github.com/lycheeverse/lychee) to check for broken links. By default it allows up to 5 redirects, as can be seen in `lychee` docs. Possible solutions would be:

1. add `max-redirects` param in [your workflow](https://github.com/alshedivat/al-folio/blob/06ca08cbaf7dbdbbe94c271df4d565c94c8b537f/.github/workflows/broken-links.yml#L52), increasing this number above 5
2. add another `--exclude-path` with the path to your page that is causing trouble
3. disable this action

**Reply by raffaelemancuso** [reply]
Thanks for the answer.

I opened [an issue at lychee](https://github.com/lycheeverse/lychee/issues/1411) since also using `--max-redirects 100` give rise to the same error.

**Reply by raffaelemancuso** [reply]
@george-gca It's bot detection which can be avoid by setting an user agent for lychee. See https://github.com/lycheeverse/lychee/issues/1411#issuecomment-2075136074.

Should we set a default user agent for lychee?

**Reply by george-gca** [reply]
I think it is a good idea. Can you send a PR for this?
